docs/markdown/Writing Plugins/common-plugin-tasks/plugins-test-goal.md

```diff
@@ -2,50 +2,247 @@
 ---
-> 📘 Example repository
-> 
-> This guide walks through adding a simple `test` implementation for Bash that runs the `shunit2` test runner. See [here](https://github.com/pantsbuild/example-plugin/blob/main/pants-plugins/examples/bash/shunit2_test_runner.py) for the final implementation.
+
 
 1. Set up a test target type
 ----------------------------
 
 Usually, you will want to add a "test" target type for your language, such as `shell_test` or `python_test`. A test target contrasts with a "source" target, such as `shell_source`. A test target is useful so that `./pants test ::` doesn't try to run tests on non-test files.
 
-When creating a test target, you should usually subclass `SingleSourceField`. You may also want to create `TimeoutField`, which should subclass `IntField`.
+When creating a test target, you should usually subclass `SingleSourceField`. You may also want to create `TimeoutField` (which should subclass `IntField`) and a `SkipField` (which should subclass `BoolField`).
 
-See [Creating new targets](doc:target-api-new-targets) for a guide on how to define new target types. 
+See [Creating new targets](doc:target-api-new-targets) for a guide on how to define new target types.
 
 ```python
 from pants.engine.target import (
     COMMON_TARGET_FIELDS,
     Dependencies,
+    BoolField,
     IntField,
     SingleSourceField,
     Target,
 )
 
-class ShellSourceField(SingleSourceField):
-    expected_file_extensions = (".sh",)
-
 
-class ShellTestSourceField(SingleSourceField):
-    pass
+class ExampleTestSourceField(SingleSourceField):
+    expected_file_extensions = (".example",)
 
 
-class ShellTestTimeoutField(IntField):
+class ExampleTestTimeoutField(IntField):
      alias = "timeout"
-     help = "Whether to time out after a certain period of time."
+     help = "Whether to time out after a certain period of time"
+
 
+class SkipExampleTestsField(BoolField):
+    alias = "skip_example_tests"
+    default = False
+    help = "If set, don't run tests on this source"
 
-class ShellTestTarget(Target):
-    alias = "bash_tests"
-    help = "Shell tests that are run via `shunit2`."
-    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, ShellTestSourceField, ShellTestTimeoutField)
+
+class ExampleTestTarget(Target):
+    alias = "example_tests"
+    help = "Example tests run by some tool"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        Dependencies,
+        ExampleTestSourceField,
+        ExampleTestTimeoutField,
+        SkipExampleTestsField,
+    )
 ```
 
 2. Set up a subclass of `TestFieldSet`
 --------------------------------------
+
+Your test-runner will need access to some / most of the fields defined on your new target to actually execute the tests within. Collect those fields into a new subclass of `TestFieldSet`, and mark at least your source field as required.
+
+If you have a "skip" field, use it in an `opt_out` method of your subclass:
+
+```python
+from pants.core.goals.test import TestFieldSet
```

Comment:
Nit, maybe add newlines like we would normally format these? Including the classmethod and the code block below

Reply:
Oops yeah will do - was copy-pasting from my 1st draft PR and I guess the newlines didn't make the trip